src/inputs/internal/MenuSearchField.tsx

```diff
@@ -0,0 +1,39 @@
+import { useTextField } from "@react-aria/textfield";
+import { useRef } from "react";
+import { Icon } from "src/components";
+import { Css, Only } from "src/Css";
+import { BeamTextFieldProps, TextFieldXss } from "src/interfaces";
+import { useTestIds } from "src/utils";
+
+interface TextFieldProps<X> extends BeamTextFieldProps<X> {
+  inlineLabel?: boolean;
+}
+
+export function MenuSearchField<X extends Only<TextFieldXss, X>>(props: TextFieldProps<X>) {
+  const tid = useTestIds(props);
+  const fieldStyles = {
+    input: {
+      ...Css.w100.mw0.outline0.bgWhite.p1.$,
+    },
+  };
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const { labelProps, inputProps } = useTextField({ ...props }, inputRef);
+
+  return (
+    <div css={Css.df.aic.fg1.bb.bGray200.$}>
+      <span css={Css.ml1.pr1.$}>
+        <Icon icon="search" />
+      </span>
+      <label {...labelProps}>{props.label}</label>
+      <input
+        {...inputProps}
```

Comment:
**question:** Could we have used `<TextField />` here? Then we'd get all the hover/focus/etc styling for free which is missing right now.

Reply:
We could, there's a very small tradeoff, unless there's a way around it. If we use `TextFieldBase` component, the border radius on the bottom left and bottom right are set at `4`. In the figma it shown as a straight line, wdyt?

Reply:
Yea, we should definitely be using the same `TextFieldBase` here. I think we live with the minor styling issues for now.

Reply:
I say that, but I think that is design's decision. I'd go back to Design and see if they are okay with this, or if they want us to rebuild it without the border-radius. We could also include a new prop to `TextFieldBase` that changes the display for this purpose. I'd like to come up with a good name for the prop that is contextual as to why we need it styled slightly differently.

Reply:
sounds good. I think design would be ok with that styling difference but let me see what they say, if it's a dealbreaker we could go down the prop route of and change the styling based off that. But lets see